src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVLinkage.java

```diff
@@ -222,12 +274,31 @@ private static int getLength(final SVCallRecord record, final int missingInserti
             return record.getLength() == null ? missingInsertionLength : Math.max(record.getLength(), 1);
         } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.BND) {
             return record.getPositionB() - record.getPositionA() + 1;
+        } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.CPX) {
+            return getLengthComplex(record, missingInsertionLength);
         } else {
             // TODO lengths less than 1 shouldn't be valid
             return Math.max(record.getLength() == null ? 1 : record.getLength(), 1);
         }
     }
 
+    /**
+     * Returns length to be used for complex variant matching
+     */
+    private static int getLengthComplex(final SVCallRecord record, final int missingInsertionLength) {
+        if (!record.isIntrachromosomal() || record.getPositionA() == record.getPositionB()) {
+            // Insertion types use the sum of cpx intervals, or the "missing" length if there are none
+            if (record.getComplexEventIntervals().isEmpty()) {
+                return missingInsertionLength;
+            } else {
+                return record.getComplexEventIntervals().stream().mapToInt(SVCallRecord.ComplexEventInterval::getLengthOnReference).sum();
```

Comment:
I think there are some issues with this length calculation. a) The intervals in CPX_INTERVALS can overlap - ie. in a dDUP type event, there can be an INV interval that is equal to the DUP interval. b) INS_iDEL events (if they should be included here - see above comment) do not contain an interval in CPX_INTERVALS that represents the source sequence, so you would only have the length of the deleted sequence, which may or may not be the desired behavior here.

I believe you could address (a) for insertion-type CPX events if you disregard any INV intervals. 

Addressing (b) depends on the desired behavior - you could look in SOURCE for the inserted sequence, but SOURCE is sometimes removed from the VCF depending on when this code is applied. But a single length measure is hard to make sense of if you're adding together the length of an inserted sequence and a deletion at the sink (ie. in an INS_iDEL or dDUP_iDEL).

Reply:
I agree, this check is probably unnecessary since we already test overlap of the individual complex intervals. I've removed this method entirely in favor of using the same check as INS types. If the variant does not have `SVLEN` assigned then `INSERTION_ASSUMED_LENGTH_FOR_OVERLAP` (50) is assumed, and the interval [POS, POS+SVLEN) is used for the overlap check.